ovos_plugin_common_play/ocp/gui.py

```diff
@@ -45,8 +47,23 @@ def bind(self, player):
                               self.handle_play_from_search)
         self.player.add_event('ovos.common_play.skill.play',
                               self.handle_play_skill_featured_media)
+        self.player.add_event('gui.page_gained_focus',
+                              self.handle_page_displayed)
         self.event_scheduler_interface = EventSchedulerInterface(name=OCP_ID, bus=self.bus)
 
+    def add_page_load_callback(self, ocp_page_id, callback, once=True):
+        self._callbacks[ocp_page_id] = (callback, once)
+
+    def handle_page_displayed(self, message):
+        skill_id = message.data.get('skill_id', "") or message.data.get('namespace', "")
+        if skill_id != OCP_ID:
+            return
+        if self._expected_ocp_page in self._callbacks:
+            callback, once = self._callbacks[self._expected_ocp_page]
+            callback()
+            if once:
```

Comment:
This should be called before the callback function and possibly with a lock so multiple messages don't trigger the function to be called more than once